src/WpBookingsUiModule.php

```diff
@@ -0,0 +1,500 @@
+
+class WpBookingsUiModule extends AbstractBaseModule
+{
+    use ContainerGetCapableTrait;
+
+    use CreateContainerExceptionCapableTrait;
+
+    use CreateNotFoundExceptionCapableTrait;
+
+    use NormalizeKeyCapableTrait;
+
+    /**
+     * Registered booking's page ID.
+     *
+     * @var
+     */
+    public $bookingsPageId;
+
+    /**
+     * Page where metabox application should be shown.
+     *
+     * @var
+     */
+    public $metaboxPageId;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $key The module key.
+     * @param string[]|Stringable[] $dependencies The module  dependencies.
+     * @param ContainerFactoryInterface $configFactory The config factory.
+     * @param ContainerFactoryInterface $containerFactory The container factory.
+     * @param ContainerFactoryInterface $compContainerFactory The composite container factory.
+     * @param EventManagerInterface $eventManager The event manager.
+     * @param EventFactoryInterface $eventFactory The event factory.
+     */
+    public function __construct(
+        $key,
+        $dependencies,
+        $configFactory,
+        $containerFactory,
+        $compContainerFactory,
+        $eventManager,
+        $eventFactory
+    )
+    {
+        $this->_initModule($key, $dependencies, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function setup()
+    {
+        return $this->_setupContainer(
+            $this->_loadPhpConfigFile(WP_BOOKINGS_UI_MODULE_CONFIG_FILE),
+            [
+                'template_manager' => function ($c) {
+                    $templateManager = new TemplateManager($this->eventManager, $this->eventFactory);
+                    $templateManager->registerTemplates($c->get('wp_bookings_ui/templates'));
+                    return $templateManager;
+                },
+                'assets_urls_map' => function ($c) {
+//                    var_dump($c->get('wp_bookings_ui/assets_urls_map_path'));die;
+                    $assetsUrlsMap = require_once $c->get('wp_bookings_ui/assets_urls_map_path');
+                    return $this->_getContainerFactory()->make([
+                        ContainerFactoryInterface::K_DATA => $assetsUrlsMap
+                    ]);
+                }
+            ]);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        $this->metaboxPageId = $c->get('wp_bookings_ui/metabox/post_type');
+
+        /* @var EventManagerInterface $eventManager */
+        $eventManager = $c->get('event_manager');
+
+        /** @var TemplateManager $templateManager */
+        $templateManager = $c->get('template_manager');
+
+        $assetsConfig = $c->get('assets_urls_map');
+
+        $eventManager->attach('admin_enqueue_scripts', function () use ($assetsConfig, $c) {

```

Comment:
Why is there a hard dependency on an event manager? Use `AbstractBaseModule#_attach()`.